Content.Shared/Trigger/Systems/ScramOnTriggerSystem.cs

```diff
@@ -60,81 +59,52 @@ protected override void OnTrigger(Entity<ScramOnTriggerComponent> ent, EntityUid
             args.Handled = true;
         }
     }
-
-    private EntityCoordinates? SelectRandomTileInRange(TransformComponent userXform, float radius)
+    /// <summary>
+    /// Method to find a random empty tile within a certain radius. Will not select off-grid tiles. Returns
+    /// null if no tile is found within a certain number of tries.
+    /// </summary>
+    private EntityCoordinates? SelectRandomTileInRange(TransformComponent userXform, float radius, int tries = 40)
     {
-        var userCoords = _transform.ToMapCoordinates(userXform.Coordinates);
-        _targetGrids.Clear();
-        _lookup.GetEntitiesInRange(userCoords, radius, _targetGrids);
-        Entity<MapGridComponent>? targetGrid = null;
-
-        if (_targetGrids.Count == 0)
-            return null;
-
-        // Give preference to the grid the entity is currently on.
-        // This does not guarantee that if the probability fails that the owner's grid won't be picked.
-        // In reality the probability is higher and depends on the number of grids.
-        if (userXform.GridUid != null && TryComp<MapGridComponent>(userXform.GridUid, out var gridComp))
-        {
-            var userGrid = new Entity<MapGridComponent>(userXform.GridUid.Value, gridComp);
-            if (_random.Prob(0.5f))
-            {
-                _targetGrids.Remove(userGrid);
-                targetGrid = userGrid;
-            }
-        }
-
-        if (targetGrid == null)
-            targetGrid = _random.GetRandom().PickAndTake(_targetGrids);
-
+        var userCoords = userXform.Coordinates;
         EntityCoordinates? targetCoords = null;
 
-        do
+        for (var i = 0; i < tries; i++)
         {
-            var valid = false;
-
-            var range = (float)Math.Sqrt(radius);
-            var box = Box2.CenteredAround(userCoords.Position, new Vector2(range, range));
-            var tilesInRange = _map.GetTilesEnumerator(targetGrid.Value.Owner, targetGrid.Value.Comp, box, false);
-            var tileList = new ValueList<Vector2i>();
-
-            while (tilesInRange.MoveNext(out var tile))
+            // We get a distance by multiplying the radius by a random float between 0 and 1,
+            // then we multiply that with a randomly angled vector.
+            // The distance is then reduced percentage-wise based on the current try count,
+            // so subsequent tries are closer and closer towards the entity. This is
+            // beneficial for smaller maps, especially when the radius is large.
+            var distance = radius * MathF.Sqrt(_random.NextFloat()) * (1 - (float)i / tries);
+            var tempTargetCoords = userCoords.Offset(_random.NextAngle().ToVec() * distance);
```

Comment:
The comments aren't accurate to the actual logic.

```suggestion
            // distance = r * sq(x) * i
            // r = the radius of the search area.
            // sq(x) = the square root of [0 - 1]. Gives a number trending to the
            // upper range of [0, 1] so that you tend to teleport further. 
            // i = A percentage based on the current try count, which results in each
            // subsequent try landing closer and closer towards the entity.
            // Beneficial for smaller maps, especially when the radius is large.
            var distance = radius * MathF.Sqrt(_random.NextFloat()) * (1 - (float)i / tries);
            
            // We then offset the user coords from a random angle * distance
            var tempTargetCoords = userCoords.Offset(_random.NextAngle().ToVec() * distance);
```